<launch>
  <arg name="y_offset" default="0" />
  <arg name="z_far" default="1.5" />
  <arg name="use_arm" default=":larm" />
  <arg name="use_kalman_filter" default="true" />
  <arg name="launch_usbcam" default="false" />
  <arg name="launch_openni" default="true" />
  <arg name="launch_creative" default="false" />
  <arg name="camera" default="camera" />
  <arg name="image" default="/camera/rgb/image_rect_color" />
  <arg name="base_frame_id" default="/base_link" />
  <arg name="debug" default="true" />
  <arg name="ns" default="psmove" />

  <group ns="$(arg ns)">
    <node name="psmove_tracker"
          pkg="jsk_2014w_rinko" type="psmove_tracker">
      <remap from="image" to="$(arg image)" />
      <remap from="psmove" to="psmove_unfiltered" if="$(arg use_kalman_filter)" />
      <param name="debug" value="$(arg debug)" />
      <param name="color_r" value="255" />
      <param name="color_b" value="255" />
      <param name="color_threshold" value="20" />
      <param name="lpf_threshold" value="3" />
      <param name="canny_threshold" value="200" />
      <param name="center_detection_threshold" value="70" />
      <param name="min_radius" value="1" />
      <param name="max_radius" value="55" />
      <param name="base_frame_id" value="$(arg base_frame_id)" />
      <param name="mirror" value="true" />
    </node>

    <node name="psmove_kalman_filter"
          pkg="jsk_2014w_rinko" type="linear_3d_kalman_filter"
          if="$(arg use_kalman_filter)">
      <remap from="pose" to="psmove_unfiltered" />
      <remap from="pose_filtered" to="psmove" />
    </node>
  </group>
</launch>
